### DISCUSSION TITLE
Machine is not reachable over network any more and 'Resource temporary out of error' in logs

### DISCUSSION THREAD

**Discussion by muhamadazmy** [cloud-hypervisor/cloud-hypervisor#2517] [Q&A] [answered]
## Setup
Host: linux kernel 5.4.10
Command:
```sh
cloud-hypervisor \
    --initramfs /var/cache/modules/flistd/mountpoint/k8s:3f8760bf9e5c5e1caa907ebe6807bcf6/k3os-initrd-amd64 \
    --memory size=2048M \
    --log-file /var/cache/modules/vmd/logs/33170-1 \
    --net tap=t-RDErEYVMoykvh fd=3,mac=ce:cb:2d:16:f8:49 \
    --disk path=/mnt/e0b57511-ca4b-4145-9586-6c1b1508e73c/vdisks/33170-1-vda,readonly=off \
    --kernel /var/cache/modules/flistd/mountpoint/k8s:3f8760bf9e5c5e1caa907ebe6807bcf6/k3os-vmlinux \
    --cmdline "console=ttyS0 reboot=k panic=1" \
    --cpus boot=1 \
    --api-socket /var/run/cloud-hypervisor/33170-1 \
    3<>/dev/tap${X}
```

In this instance of CH we pass 2 networking devices
- the first one is a tap device.
- the second one is a macvtap device passed as an open file descriptor. This setup has been working for a while with other instances.
- This VM itself was running good for over 7 days

Now it can't be reached over the macvtap interface (but over the tap is okay). The following logs can be found in the logs.
## Error Logs
```
cloud-hypervisor: 430168.751545616s: <_net2_qp0> ERROR:net_util/src/queue_pair.rs:70 -- net: tx: failed writing to tap: Resource temporarily unavailable (os error 11)
cloud-hypervisor: 430168.751928225s: <_net2_qp0> ERROR:virtio-devices/src/net.rs:181 -- Error processing TX queue: NetQueuePair(WriteTap(Os { code: 11, kind: WouldBlock, message: "Resource temporarily unavailable" }))
```

Note:
the reason we using `macvtap` instead of a `tap` for the 2nd device is this issue here https://github.com/cloud-hypervisor/cloud-hypervisor/discussions/2515 which we didn't figure out yet 

> Suggestion: Is retrying a possibility ? or at least bubbling up an error to host but the next operation should be processed, and not completely stopping the device controller.

**Comment by rbradford** [comment, marked as answer]
See #2518 for a fix

**Reply by muhamadazmy** [reply]
Thank you, will give it a try and let you know. Thanks again

**Reply by muhamadazmy** [reply]
Seems to be working so far.Any plans for a release with this fix?

**Reply by sboeuf** [reply]
We're releasing this week on Thursday, the fix will be included :)
